streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -356,20 +358,24 @@ public String queryableStoreName() {
     @Override
     public KTable<K, V> suppress(final Suppressed<K> suppressed) {
         final String name = builder.newProcessorName(SUPPRESS_NAME);
+        final String storeName = builder.newStoreName(SUPPRESS_NAME);
 
         final ProcessorSupplier<K, Change<V>> suppressionSupplier =
             () -> new KTableSuppressProcessor<>(
                 buildSuppress(suppressed),
+                storeName,
                 keySerde,
                 valSerde == null ? null : new FullChangeSerde<>(valSerde)
             );
 
-        final ProcessorParameters<K, Change<V>> processorParameters = new ProcessorParameters<>(
-            suppressionSupplier,
-            name
-        );
 
-        final ProcessorGraphNode<K, Change<V>> node = new ProcessorGraphNode<>(name, processorParameters, false);
+        final ProcessorGraphNode<K, Change<V>> node = new StatefulProcessorNode<>(
+            name,
+            new ProcessorParameters<>(suppressionSupplier, name),
+            null,
```

Comment:
Why not put the storeName here? In that case the processor will not be connected to the state store.

Reply:
I had a little confusion about that... Supplying the name there results in an exception.

Apparently, we are supposed to pass in the names only if the store has been pre-registered. Since were're creating the store right now, we pass in the store builder on the next line, and the topology build process will invoke the builder and connect it to the processor.

The state store and processor get connected either way.

Or did I misunderstand what you were suggesting?

Reply:
You are actually right. NVM on the above comment.